src/OpenTelemetry.Extensions.Hosting/OpenTelemetryServicesExtensions.cs

```diff
@@ -85,5 +114,36 @@ private static IServiceCollection AddOpenTelemetryTracing(this IServiceCollectio
 
             return services;
         }
+
+        /// <summary>
+        /// Adds OpenTelemetry MeterProvider to the specified <see cref="IServiceCollection" />.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <param name="createMeterProvider">A delegate that provides the tracer provider to be registered.</param>
+        /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+        private static IServiceCollection AddOpenTelemetryMetrics(this IServiceCollection services, Func<IServiceProvider, MeterProvider> createMeterProvider)
+        {
+            if (services is null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
```

Comment:
Maybe it would be better to add this null guard to public methods?

Reply:
I'm going to vote to leave as-is on this one. Reasoning: `services` is the thing being extended. It is unlikely to be `null` unless someone is going out of their way to call it directly AND pass `null`. So given that should be rare, having it here saves having to duplicate the check in all the publics. 2 checks instead of 4, basically.